yarn-project/noir-contracts/src/contracts/uniswap_contract/src/util.nr

```diff
@@ -0,0 +1,83 @@
+use dep::std::hash::{pedersen_with_separator, sha256};
+use dep::aztec::constants_gen::GENERATOR_INDEX__SIGNATURE_PAYLOAD;
+
+fn compute_message_hash<N>(args: [Field; N]) -> Field {
+  // @todo @lherskind We should probably use a separate generator for this, 
+  //                  to avoid any potential collisions with payloads.
+  pedersen_with_separator(args, GENERATOR_INDEX__SIGNATURE_PAYLOAD)[0]
+}
+
+// This method computes the L2 to L1 message content hash for both private and public flow. 
+// refer `l1-contracts/test/portals/UniswapPortal.sol` on how L2 to L1 message is expected
+// for private flow, instead of specifying who is recipient on L2 will be, we specify the secret hash for redeeming minted notes
+fn compute_swap_content_hash(
```

Comment:
As mentioned earlier I think clarity is reduced with the ifs in here as inputs have multiple meanings.